js/shopping_list_test.js

```diff
@@ -0,0 +1,118 @@
+    expect(someName.name).to.be.a('string');
+  });
+  it('Should have a property of description', function(){
+    expect(someName).to.have.ownPropertyDescriptor('description');
+  });
+  it('Description property should be a string', function(){
+    expect(someName.description).to.be.a('string');
+  });
+  it('Should have a property of is_done', function(){
+    expect(someName).to.have.ownPropertyDescriptor('is_done');
+  });
+  it('is_done property should be a boolean', function(){
+    expect(someName.is_done).to.be.a('boolean');
+  });
+  it('Should have a method named check', function(){
+    expect(someName.check).to.be.a('function');
+  });
+  it('Should reassign the value of is_done to true', function(){
+    someName.check();
+    expect(someName.is_done).to.be.true;
+  });
+
+  it('Should have a method named uncheck', function(){
+    expect(someName.uncheck).to.be.a('function');
+  });
+  it('Should reassign the value of is_done to false', function(){
+    someName.uncheck();
+    expect(someName.is_done).to.be.false;
+  });
+  it('Should have a method named render', function(){
+    expect(someName.render).to.be.a('function');
+  });
+  it('Should return a string', function(){
+    expect(someName.render()).to.be.a('string');
+  });
+  it('It should return a li with a class of "completed_is_done"', function(){
+    expect(someName.render()).contains('<li class="completed_' + someName.is_done + '">');
+  });
+  it('It should return a <span> with the list item name', function(){
+    expect(someName.render()).contains('<span>' + someName.name + '</span>');
+  });
+  it('It should return a <span> with the item description', function(){
+    expect(someName.render()).contains('<span>' + someName.description + '</span>');
+  });
+});
+
+describe('Shopping List Class', function(){
+  let list = ShoppingList;
+  let someList = null;
+
+  before(function(){
+    someList = new ShoppingList();
+    someItem = new ShoppingListItem('carrots', 'a vegetable');
+  });
+
+  it('Shopping list is a class', function(){
+    expect(list).to.be.a('function');
+  });
+  it('Should have a property of items', function(){
+    expect(someList).to.have.ownPropertyDescriptor('items');
+  });
+
+  it('Should have a method addItem', function(){
+    expect(someList.addItem).to.be.a('function');
+  });
+  it('Should take a ShoppingListItem object', function() {
+    someList.addItem(someItem);
+    expect(someItem).to.be.a.instanceOf(ShoppingListItem);
+  });
+
+  it('Should throw an error if not a ShoppingListItem', function(){
+    expect(someList.addItem.bind(this)).to.throw(TypeError);
+  });
+  it('Should have a method removeItem', function(){
+    expect(someList.removeItem).to.be.a('function');
+  });
+  it('removeItem should take a ShoppingListItem object or no argument', function() {
+    someList.removeItem(someItem);
+    expect(someItem).to.be.a.instanceOf(ShoppingListItem) || expect(someItem).to.be.undefined;
```

Comment:
I am not sure why you would do an or test on an assertion.
If invoking removeItem has two possible results you should create two tests.